added support for the LoFive R1

CobaltCause opened the pull request:
Note the comment at `build.rs:37`! I'm also not sure if I've changed *everything* necessary for correct support of the R1. I did actually test this on real hardware though, and it seems to work.

Commit:
added support for the LoFive R1

Disasm commented on src/clock.rs:
```diff
@@ -6,11 +6,15 @@ use e310x_hal::{
-/// For HiFive1 and HiFive1 Rev B boards external oscillators are enabled for
-/// both high-frequency and low-frequency clocks.
+/// For HiFive1, HiFive1 Rev B, and LoFive R1 boards, the external oscillators
+/// are enabled for both high-frequency and low-frequency clocks.
```
I believe that LoFive R1 does not have any external oscillator for the low-frequency clock:
https://raw.githubusercontent.com/wiki/mwelling/lofive/lofive-r1-schematic.png

CobaltCause replied:
Oh, so should I have added the R1 to the docs on the other function then? Honestly, I just assumed that it'd be grouped with the HiFive1 Rev B because they both use the G002...

Commit:
corrected src/clock.rs

Commit:
added CI rule for `board-lofive-r1` feature

Commit:
added linkerscript for the LoFive R1

Note that this linkerscript assumes the bootloader is installed.
Instructions on how to do that can be found here:

https://github.com/mwelling/lofive/wiki#programming-to-lofive-r1-using-the-new-sdk

It'll "work" without the bootloader, but user code won't start after a
reset.
